feature/experimental/bgp/otg_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,534 @@
+	}
+
+	// Check last established timestamp
+	lestTime := gnmi.Get(t, dut, nbrPath.State()).GetLastEstablished()
+	t.Logf("BGP last est time :%v, flapTime :%v", lestTime, lastFlapTime)
+	if lestTime < lastFlapTime {
+		t.Errorf("Bad last-established timestamp: got %v, want >= %v", lestTime, lastFlapTime)
+	}
+
+	// Check BGP Transitions
+	nbr := gnmi.Get(t, dut, statePath.State()).GetNeighbor(ateAttrs.IPv4)
+	estTrans := nbr.GetEstablishedTransitions()
+	t.Logf("Got established transitions: %d", estTrans)
+	if estTrans != 1 {
+		t.Errorf("Wrong established-transitions: got %v, want 1", estTrans)
+	}
+
+	// Check BGP neighbor address from telemetry
+	addrv4 := gnmi.Get(t, dut, nbrPath.State()).GetNeighborAddress()
+	t.Logf("Got ipv4 neighbor address: %s", addrv4)
+	if addrv4 != ateAttrs.IPv4 {
+		t.Errorf("BGP v4 neighbor address: got %v, want %v", addrv4, ateAttrs.IPv4)
+	}
+	// Check BGP neighbor address from telemetry
+	peerAS := gnmi.Get(t, dut, nbrPath.State()).GetPeerAs()
+	if peerAS != ateAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, ateAS)
+	}
+
+	// Check BGP neighbor is enabled
+	if !gnmi.Get(t, dut, nbrPath.State()).GetEnabled() {
+		t.Errorf("Expected neighbor %v to be enabled", ateAttrs.IPv4)
+	}
+}
+
+// Function to configure ATE configs based on args and returns ate topology handle.
+func configureATE(t *testing.T, ateParams *bgpTestParams, connectionType connType, auth authType) gosnappi.Config {
+	t.Helper()
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := gosnappi.NewConfig()
+
+	topo.Ports().Add().SetName(port1.ID())
+	dev := topo.Devices().Add().SetName(ateAttrs.Name)
+	eth := dev.Ethernets().Add().SetName(ateAttrs.Name + ".Eth")
+	eth.Connection().SetPortName(port1.ID())
+	eth.SetMac(ateAttrs.MAC)
+
+	ip := eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ip.SetAddress(ateAttrs.IPv4).SetGateway(dutAttrs.IPv4).SetPrefix(uint32(ateAttrs.IPv4Len))
+
+	bgp := dev.Bgp().SetRouterId(ateAttrs.IPv4)
+	peerBGP := bgp.Ipv4Interfaces().Add().SetIpv4Name(ip.Name()).Peers().Add()
+	peerBGP.SetName(ateAttrs.Name + ".BGP4.peer")
+	peerBGP.Advanced().SetHoldTimeInterval(ateHoldTime)
+	if auth == md5Auth {
+		peerBGP.Advanced().SetMd5Key(authPassword)
+	}
+	peerBGP.SetPeerAddress(ip.Gateway()).SetAsNumber(uint32(ateParams.localAS))
+	if connectionType == connInternal {
+		peerBGP.SetAsType(gosnappi.BgpV4PeerAsType.IBGP)
+	} else {
+		peerBGP.SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+	}
+
+	return topo
+
+}
+
+// createCeaseAction creates the BGP cease notification action in gosnappi
+func createCeaseAction(t *testing.T) gosnappi.ControlAction {
+	t.Helper()
+	ceaseAction := gosnappi.NewControlAction()
+	ceaseAction.Protocol().SetChoice(gosnappi.ActionProtocolChoice.BGP).Bgp().
+		SetChoice(gosnappi.ActionProtocolBgpChoice.NOTIFICATION).
+		Notification().SetNames([]string{}).
+		SetChoice(gosnappi.ActionProtocolBgpNotificationChoice.CUSTOM).
```

Comment:
These `SetChoices` should be removed as well.

Reply:
This can actually now be simplified to: `ceaseAction.Protocol().Bgp().Notification().SetNames([]string{ateAttrs.Name + ".BGP4.peer"}).Cease()`

Making the peer name explicit improves clarity as well.

Reply:
Made the suggested changes and the test passes on hw too now